### ISSUE TITLE
Resistors For USB Host

### ISSUE THREAD

**Issue by Norto04** [sekigon-gonnoc/Pico-PIO-USB#101]
Does this project follow the USB Specification for the host implementation? Im having trouble understanding if it requires 15k (changed to 14.5k - 24.8k) resistors and if there something that needs to be disconnected after the device negotiates what speed its using. Or does PIO USB implement this somewhere I havent seen? There is a resistor pdf in the link below.

https://www.usb.org/sites/default/files/usb_20_20230224.zip

P.S, What are the 22ohm resistors mentioned in the README used for?

**Comment by drojf** [comment]
I was wondering too. I don't have a proper answer, but I saw this application note describing these 22 ohm resistors and thought it might be useful: https://www.onsemi.com/pub/collateral/and8074-d.pdf

I think the ~~15K pull up and 1.5K pull down~~ are for indicating upstream/downstream and low speed/full speed. EDIT: it should be 15K pull **down** for downstream and 1.5K pull **up** for upstream.

I'm not sure how these can be omitted (See [Adafruit's RP2040 USB Host Schematic with only 22ohm resistors](https://learn.adafruit.com/assets/120409)) but I guess it works anyway.

**Comment by Dowster64** [comment]
Just in case anyone is watching this.  I am using the Waveshare RP2350-USB-A dev board.  As delivered it does not work properly with this library in host mode.  This blog post articulates that:
https://qsantos.fr/2025/11/21/fixing-the-rp2350-usb-a-not-working-as-usb-host/
The schematics referenced are very clear, but you have to read the comments section for a correct answer.  The 15k resistors are added as pull down resistors (to ground) on the board (I added them to the terminals of the USB port).  the existing 1k5 pull up (so the board can act as high speed device) is removed.  Note the device you plug into the host has a pull up resistor either on D+ or D- to indicate what type of device it is.
To summarise, without the 15k pull down resistors the board can't be a host as it can't tell if devices are plugged in or out. ie the mount, umount callbacks won't run AND you need the device you plug in to have the pull up resistor.  IF you leave the 1k5 pull up it will always think a high speed device is mounted.

When you have these physical resistors in place, the library works as expected.  Another way of saying that is that if your hardware follows the USB standard (the link above also references [usb in a nutshell](https://www.beyondlogic.org/usbnutshell/usb2.shtml)) , then this library will work (also according to the examples the clock speed needs to be the correct multiple too)
